### ISSUE TITLE
Dialog: Ensure focus position starts from the Dialog content

### ISSUE THREAD

**Issue by kohli-pratik-sage** [Sage/carbon#4975]
### Desired behaviour

When the dialog component is opened, the keyboard focus position begins from the cross close button despite the body of the dialog containing focusable elements.

### Current behaviour

When the dialog component is opened and it contains focusable elements within the body, the keyboard focus position bypasses the cross close button and begins from the first focusable element within the body.

### Suggested Solution

Adding a `tabindex="-1"` attribute to the dialog element should ensure the focus cannot return to it which would mean screen reader users start reading from the start of the content, and keyboard users would get to the first interactive control (cross close button) by pressing the tab key.

### CodeSandbox or Storybook URL

https://codesandbox.io/s/objective-butterfly-qwk6b8?file=/src/index.js

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by nicktitchmarsh** [comment]
FE-5092

**Comment by nicktitchmarsh** [comment]
Comment from @tempertemper 

Ideally the default focus position would be on the dialog element itself, via a tabindex="-1" attribute, so that focus couldn’t return to it. That would mean screen reader users start reading from the start of the content, and keyboard users would get to the first interactive control (close button probably) by pressing the tab key.
It’s ok for focus to be placed on a form input or a ‘confirm my choice’ type button, but the exception rather than the rule: it should only happen when the content above the control doesn’t add any more info.
